### PR TITLE
[CTCP-3347]  updated the get all endpoints to fetch totalCount

### PR DESCRIPTION
**Pull request by prashanthmamidi**

### CODE REVIEW

File: app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala [hunk 1]
```diff
@@ -247,17 +255,19 @@ class MovementsRepositoryImpl @Inject() (
...
       )
 
-    mongoRetry(Try(collection.aggregate[MessageResponse](aggregates)) match {
+    mongoRetry(Try(collection.aggregate[PaginationMessageSummary](aggregates)) match {
       case Success(obs) =>
         obs
           .toFuture()
           .map(
-            response => Right(response.toVector)
+            response => Right(response.head)
```
Comment (derekspalmer) [inline app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala:270]: Why just the head?

File: app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala [hunk 2]
```diff
@@ -223,9 +231,9 @@ class MovementsRepositoryImpl @Inject() (
     page: Option[PageNumber] = None,
     count: Option[ItemCount] = None,
     receivedUntil: Option[OffsetDateTime] = None
-  ): EitherT[Future, MongoError, Vector[MessageResponse]] = {
+  ): EitherT[Future, MongoError, PaginationMessageSummary] = {
 
-    val projection = MessageResponse.projection
+    val messageSummaryQuery = Json.obj("$ifNull" -> Json.arr("$messageSummary", "[]"))
```
Comment (derekspalmer) [inline app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala:236]: Not sure if this is the best way ... but guess it works.

File: app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala [hunk 3]
```diff
@@ -325,26 +335,26 @@ class MovementsRepositoryImpl @Inject() (
...
 
-    mongoRetry(Try(collection.aggregate[MovementWithoutMessages](aggregates)) match {
+    mongoRetry(Try(collection.aggregate[PaginationMovementSummary](aggregates)) match {
       case Success(obs) =>
         obs
           .toFuture()
           .map {
-            response =>
-              Right(response.toVector)
+            response => Right(response.head)
```
Comment (derekspalmer) [inline app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala:357]: Same here.

**Review by drnaylor-oc — APPROVED** [review]
I know that regression tests were run with this so I won't do it myself. I'm happy with this (and I have to say, good job with that mongo query -- I'd have probably just made two calls!)

**Review by derekspalmer — APPROVED** [review]

File: app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala (hunk 3 above)
Comment (drnaylor-oc) [reply app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala:357]: Same as above

File: app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala (hunk 1 above)
Comment (prashanthmamidi) [reply app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala:270]: yes

File: app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala (hunk 2 above)
Comment (prashanthmamidi) [reply app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala:236]: yes,  it works so I went with this

**Pull request merged**